esp32-controller/src/handler/wsEventHandler.cpp

```diff
@@ -0,0 +1,58 @@
+#include "wsEventHandler.h"
+
+extern EngineController engine_controller;
+
+void wsEventHandler(void* event_handler_arg, esp_event_base_t event_base,
+                    int32_t event_id, void* event_data) {
+  esp_websocket_event_data_t data = *(esp_websocket_event_data_t*)event_data;
+  switch (event_id) {
+    case WEBSOCKET_EVENT_CONNECTED: {
+      Serial.println("Websocket connected");
+      break;
+    }
+    case WEBSOCKET_EVENT_DISCONNECTED: {
+      Serial.println("Websocket disconnected");
+      break;
+    }
+    case WEBSOCKET_EVENT_DATA: {
+      uint16_t message_type =
+          ((uint16_t)data.data_ptr[0] << 8) | data.data_ptr[1];
+      uint16_t parameter[4];
+      for (uint8_t i = 1; i < 5; i++) {
+        parameter[i] =
+            ((uint16_t)data.data_ptr[2 * i] << 8) | data.data_ptr[(2 * i) + 1];
+      }
+
+      uint16_t duration = str_to_number<uint16_t, uint16_t>(parameter, 4);
```

Comment:
Maybe we should add comments to this block, or abstract the arguments parsing to a separate function.
Personally speaking, the amount of loops and type casts make the code a little bit confusing.

Reply:
It is only a temporary code, I would like to test it and discuss it a bit more before committing to the solution.